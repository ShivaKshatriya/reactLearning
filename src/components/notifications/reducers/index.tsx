import { combineReducers } from 'redux';
import {
    Action,
    DISMISS_NOTIFICATION,
    CLEAR_NOTIFICATIONS
} from '../actions';

export namespace Store {
    export interface Notification {
        id: number;
        text: string;
        createdOn: number;
    };

    export interface All {
        notifications: Notification[]
    };
}

function notifications(state: Store.Notification[] = [], action: Action): Store.Notification[] {
    switch (action.type) {
        case DISMISS_NOTIFICATION:
            let notificationsNew = state.filter((notification) => {
                return notification.id !== action.id;
            });
            return notificationsNew;
        case CLEAR_NOTIFICATIONS:
            return [];

        default: return state;
    }
}

export const reducers = combineReducers<Store.All>({
    notifications
});