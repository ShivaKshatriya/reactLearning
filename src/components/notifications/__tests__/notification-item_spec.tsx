import * as React from 'react';
import { Store } from '../reducers';
import NotificationItem from '../notification-item';
import { create } from 'react-test-renderer';

describe('component/notification-item', () => {
    it('renders correctly', () => {
        let sampleNotification = {
            id: 0,
            text: 'Test String'
        } as Store.Notification;

        let dismissDummy = function() {

        }

        const tree = create(
            <NotificationItem notification={sampleNotification} dismiss={dismissDummy} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
