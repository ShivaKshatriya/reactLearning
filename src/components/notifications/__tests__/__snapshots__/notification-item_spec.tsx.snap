exports[`component/notification-item renders correctly 1`] = `
<li
  className="notification-item">
  Test String
  <button
    className="button"
    onClick={[Function]}>
    Dismiss
  </button>
</li>
`;
